### ISSUE TITLE
Cannot compile on macOS 13 Beta 3

### ISSUE THREAD

**Issue by spnda** [KhronosGroup/MoltenVK#1634]
```
/Users/sean/Documents/git/spnda/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm:1692:8: error: 
      'MTLLanguageVersion1_0' is unavailable: not available on macOS
                case MTLLanguageVersion1_0:
```
I'm unsure if it's possible to expand the relevant preprocessor check to always exclude those lines when compiling for macOS, as I am not sure if older Metal devices did actually exclusively support MSL 1.0. Therefore, the following diff might already be the answer.
```diff
-#if MVK_IOS_OR_TVOS || MVK_XCODE_14
+#if MVK_IOS_OR_TVOS || (MVK_XCODE_14 && !MVK_MACOS)
         case MTLLanguageVersion1_0:
             setMSLVersion(1, 0);
             break;
 #endif
```

**Comment by billhollings** [comment]
Actually...the `MVK_XCODE_14` was only recently added because Xcode 14 Beta 2 incorrectly added support for `MTLLanguageVersion1_0` in macOS.

Probably best to simply revert and remove that additional macro. I'll grab Xcode 14 Beta 3 and make that change.

**Comment by billhollings** [comment]
Fixed in PR #1641. Please retest with latest code and close this issue if it fixes the problem you encountered.
